internal/server/recon/recognize.go

```diff
@@ -446,24 +455,67 @@ func getNumSpansForContainedIn(spans []*pb.TokenSpans_Span, startIdx int) int {
 	return 0
 }
 
-func combineContainedInSingle(
-	spans []*pb.TokenSpans_Span, startIdx, numSpans int) *pb.TokenSpans_Span {
+func combineTokenSpans(spans []*pb.TokenSpans_Span, startIdx, numSpans int) *pb.TokenSpans_Span {
 	startSpan := spans[startIdx]
-	endSpan := spans[startIdx+numSpans-1]
-
+	
 	res := &pb.TokenSpans_Span{Tokens: startSpan.Tokens}
 	for i := 1; i < numSpans; i++ {
 		res.Tokens = append(res.Tokens, spans[startIdx+i].GetTokens()...)
 	}
 
-	// This map is used to collect all the places for the combined span, with dedup.
+	return combineContainedInTokens(res, spans, startIdx, numSpans)
+}
+
+func getNextPlaceTokenSpan(spans []*pb.TokenSpans_Span, startIdx int) (*pb.TokenSpans_Span, int) {
+	for i := startIdx; i < len(spans); i++ {
+		if len(spans[i].GetPlaces()) > 0 {
+			return spans[i], i
+		}
+	}
+	return nil, -1
+}
+
+func findRecogPlaces(startSpan, nextSpan *pb.TokenSpans_Span) (map[string]*pb.RecogPlace, []string) {
```

Comment:
nit: can we add a comment explaining what this function is doing since the name doesn't exactly explain it?

Reply:
Done.